### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by peterserfozo**

### CODE REVIEW

File: tests/src/Functional/ConfigurationPermissionTest.php [hunk 1]
```diff
@@ -104,42 +84,35 @@ protected function assertPaths(bool $access) {
...
-      list($run_path, $run_query) = $this->findLink('Now');
-      $visit_path($schedule_path, $schedule_query);
-      $visit_path($run_path, $run_query);
-    }
-    else {
-      $visit_path('/admin/config/apigee-edge/sync/schedule');
-      $visit_path('/admin/config/apigee-edge/sync/run');
+    // Get all routes defined by the module and check every route that requires
+    // the permission "administer apigee edge".
+    $module_path = $this->container->get('module_handler')->getModule('apigee_edge')->getPath();
```
Comment (mxr576) [inline tests/src/Functional/ConfigurationPermissionTest.php:89]: Wooot!!! :) Cool, I never thought that my idea could be solved this easily in Drupal 8.

**Review by mxr576 — APPROVED** [review]

**Review by mxr576 — CHANGES_REQUESTED** [review]
"TypeError: Argument 3 passed to Apigee\Edge\Api\Management\Controller\DeveloperController::__construct() must be of the type array, null given, called in /opt/drupal-module/src/Entity/Controller/DeveloperController.php on line 63"

https://travis-ci.org/peterserfozo/apigee-edge-drupal/builds/414493907

File: modules/apigee_edge_apiproduct_rbac/src/RoleBasedAccessSettingsBatch.php [hunk 2]
```diff
@@ -33,27 +33,27 @@
...
-   * @param string|null $originalAttributeName
+   * @param string|null $original_attribute_name
    *   Name of the attribute that originally stored the role assignments.
    * @param array $context
    *   Batch context.
    *
    * @see callback_batch_operation()
    */
-  public static function batchOperation(array $productNameDisplayNameMap, array $productNameRidsMap, string $attributeName, string $originalAttributeName, array &$context) : void {
+  public static function batchOperation(array $produc_name_display_name_map, array $product_name_rids_map, string $attribute_name, string $original_attribute_name, array &$context): void {
```
Comment (tamasd) [inline modules/apigee_edge_apiproduct_rbac/src/RoleBasedAccessSettingsBatch.php]: Typo

File: src/Entity/Controller/DeveloperAppCredentialController.php [hunk 3]
```diff
@@ -44,111 +44,93 @@ class DeveloperAppCredentialController extends EdgeDeveloperAppCredentialControl
...
   /**
    * {@inheritdoc}
-   *
-   * @throws \Drupal\Core\TempStore\TempStoreException
    */
-  public function create(string $consumerKey, string $consumerSecret): AppCredentialInterface {
-    $credential = parent::create($consumerKey, $consumerSecret);
+  public function create(string $consumer_key, string $consumer_secret): AppCredentialInterface {
+    $credential = parent::create($consumer_key, $consumer_secret);
     \Drupal::service('event_dispatcher')->dispatch(AppCredentialCreateEvent::EVENT_NAME, new AppCredentialCreateEvent(AppCredentialCreateEvent::APP_TYPE_DEVELOPER, $this->developerId, $this->appName, $credential));
```
Comment (tamasd) [inline src/Entity/Controller/DeveloperAppCredentialController.php:50]: Dependency injection?

File: src/Entity/Controller/DeveloperAppCredentialController.php [hunk 4]
```diff
@@ -44,111 +44,93 @@ class DeveloperAppCredentialController extends EdgeDeveloperAppCredentialControl
...
+    array $api_products,
+    AttributesProperty $app_attributes,
+    string $callback_url,
     array $scopes = [],
-    string $keyExpiresIn = '-1'
+    string $key_expires_in = '-1'
   ): AppCredentialInterface {
-    $credential = parent::generate($apiProducts, $appAttributes, $callbackUrl, $scopes, $keyExpiresIn);
+    $credential = parent::generate($api_products, $app_attributes, $callback_url, $scopes, $key_expires_in);
     \Drupal::service('event_dispatcher')->dispatch(AppCredentialGenerateEvent::EVENT_NAME, new AppCredentialGenerateEvent(AppCredentialCreateEvent::APP_TYPE_DEVELOPER, $this->developerId, $this->appName, $credential));
```
Comment (tamasd) [inline src/Entity/Controller/DeveloperAppCredentialController.php:66]: Dependency injection?

File: src/Entity/Controller/DeveloperAppCredentialController.php [hunk 5]
```diff
@@ -44,111 +44,93 @@ class DeveloperAppCredentialController extends EdgeDeveloperAppCredentialControl
...
    * {@inheritdoc}
-   *
-   * @throws \Drupal\Core\TempStore\TempStoreException
    */
-  public function addProducts(string $consumerKey, array $apiProducts): AppCredentialInterface {
-    $credential = parent::addProducts($consumerKey, $apiProducts);
-    \Drupal::service('event_dispatcher')->dispatch(AppCredentialAddApiProductEvent::EVENT_NAME, new AppCredentialAddApiProductEvent(AppCredentialCreateEvent::APP_TYPE_DEVELOPER, $this->developerId, $this->appName, $credential, $apiProducts));
+  public function addProducts(string $consumer_key, array $api_products): AppCredentialInterface {
+    $credential = parent::addProducts($consumer_key, $api_products);
+    \Drupal::service('event_dispatcher')->dispatch(AppCredentialAddApiProductEvent::EVENT_NAME, new AppCredentialAddApiProductEvent(AppCredentialCreateEvent::APP_TYPE_DEVELOPER, $this->developerId, $this->appName, $credential, $api_products));
```
Comment (tamasd) [inline src/Entity/Controller/DeveloperAppCredentialController.php:76]: Dependency injection?

File: src/Entity/Controller/DeveloperAppCredentialController.php [hunk 6]
```diff
@@ -44,111 +44,93 @@ class DeveloperAppCredentialController extends EdgeDeveloperAppCredentialControl
...
   /**
    * {@inheritdoc}
-   *
-   * @throws \Drupal\Core\TempStore\TempStoreException
    */
-  public function delete(string $entityId): AppCredentialInterface {
-    $entity = parent::delete($entityId);
+  public function delete(string $entity_id): AppCredentialInterface {
+    $entity = parent::delete($entity_id);
     \Drupal::service('event_dispatcher')->dispatch(AppCredentialDeleteEvent::EVENT_NAME, new AppCredentialDeleteEvent(AppCredentialCreateEvent::APP_TYPE_DEVELOPER, $this->developerId, $this->appName, $entity));
```
Comment (tamasd) [inline src/Entity/Controller/DeveloperAppCredentialController.php:134]: Dependency injection?

**Comment by mxr576** [comment]
My test build on Travis: https://travis-ci.org/mxr576/apigee-devportal-drupal/jobs/415309939

File: tests/src/Functional/CacheTest.php [hunk 7]
```diff
@@ -58,91 +74,53 @@ protected function setUp() {
...
+    $this->userDeletedTest();
   }
 
   /**
    * Tests that credentials are not cached, but found on the app page.
    */
-  public function testCredentials() {
-    $this->drupalGet("/user/{$this->account->id()}/apps/{$this->app->getName()}");
+  protected function credentialsTest() {
+    $this->drupalGet("/user/{$this->account->id()}/apps/{$this->developerApp->getName()}");
```
Comment (tamasd) [inline tests/src/Functional/CacheTest.php]: Shouldn't this also use `Url::fromRoute()`?

**Review by tamasd — CHANGES_REQUESTED** [review]

**Review by tamasd — APPROVED** [review]

File: src/Entity/Controller/DeveloperAppCredentialController.php (hunk 3 above)
Comment (mxr576) [reply src/Entity/Controller/DeveloperAppCredentialController.php:50]: No DI in controller classes, unless you want to pass it from the storage in the constructor of this class. IIRC for some reasons I dropped this idea before.

**Review by mxr576 — APPROVED** [review]

**Pull request merged**